### PR TITLE
fix(refresher): import icons to avoid errors in react and vue

### LINKED ISSUE #24480: bug: Uncaught TypeError: Failed to construct 'URL': Invalid base URL
### Prerequisites

- [X] I have read the [Contributing Guidelines](https://github.com/ionic-team/ionic-framework/blob/main/.github/CONTRIBUTING.md#creating-an-issue).
- [X] I agree to follow the [Code of Conduct](https://ionicframework.com/code-of-conduct).
- [X] I have searched for [existing issues](https://github.com/ionic-team/ionic-framework/issues) that already report this problem, without success.

### Ionic Framework Version

- [ ] v4.x
- [ ] v5.x
- [X] v6.x

### Current Behavior

The following message shows on the browser console whenever I render the component that contains `IonRefresherContent`.

```
Uncaught TypeError: Failed to construct 'URL': Invalid base URL
    at getAssetPath (index.js:2680)
    at getNamedUrl (icon.js:46)
    at getUrl (icon.js:27)
    at HTMLElement.loadIcon (icon.js:255)
    at icon.js:227
    at HTMLElement.waitUntilVisible (icon.js:250)
    at HTMLElement.connectedCallback (icon.js:225)
    at HTMLElement.connectedCallback (index.js:2280)
    at addVnodes (index.js:803)
    at patch (index.js:991)
    at renderVdom (index.js:1187)
    at callRender (index.js:1449)
    at updateComponent (index.js:1371)
    at index.js:1352
    at then (index.js:1582)
    at dispatchHooks (index.js:1352)
    at Array.dispatch (index.js:1320)
    at consume (index.js:2954)
    at flush (index.js:3009)
```

### Expected Behavior

Render `IonRefresherContent` without error messages.

### Steps to Reproduce

1. Use the following combo anywhere.

```
<IonRefresher slot="fixed">
  <IonRefresherContent />
</IonRefresher>
```

2. The error message is displayed on the browser console.

### Code Reproduction URL

_No response_

### Ionic Info

```
Ionic:

   Ionic CLI       : 6.18.1 (/Users/ken.zhang/.nvm/versions/node/v13.10.1/lib/node_modules/@ionic/cli)
   Ionic Framework : @ionic/react 6.0.1

Capacitor:

   Capacitor CLI      : 3.0.0
   @capacitor/android : not installed
   @capacitor/core    : 3.0.0
   @capacitor/ios     : not installed

Utility:

   cordova-res (update available: 0.15.4) : 0.15.3
   native-run                             : 1.5.0

System:

   NodeJS : v13.10.1 (/Users/ken.zhang/.nvm/versions/node/v13.10.1/bin/node)
   npm    : 7.14.0
   OS     : macOS Big Sur
```

### Additional Information

```
dependencies in package.json

"@capacitor/app": "1.0.0",
"@capacitor/camera": "1.0.0",
"@capacitor/core": "3.0.0",
"@capacitor/filesystem": "1.0.0",
"@capacitor/haptics": "1.0.0",
"@capacitor/keyboard": "1.0.0",
"@capacitor/storage": "1.0.0",
"@ionic-native/camera": "5.36.0",
"@ionic-native/clipboard": "5.36.0",
"@ionic-native/core": "5.36.0",
"@ionic-native/firebase": "5.36.0",
"@ionic/pwa-elements": "3.1.1",
"@ionic/react": "6.0.1",
"@ionic/react-router": "6.0.1",
"@react-google-maps/api": "2.2.0",
"@reduxjs/toolkit": "1.5.1",
"@testing-library/jest-dom": "5.11.9",
"@testing-library/react": "11.2.5",
"@testing-library/user-event": "12.6.3",
"@types/deep-equal": "1.0.1",
"@types/jest": "26.0.20",
"@types/node": "12.19.15",
"@types/prettier": "2.2.3",
"@types/react": "16.14.3",
"@types/react-dom": "16.9.10",
"@types/react-geocode": "0.2.0",
"@types/react-redux": "7.1.16",
"@types/react-router": "5.1.11",
"@types/react-router-dom": "5.1.7",
"@types/styled-components": "5.1.9",
"@types/uuid": "8.3.3",
"connected-react-router": "6.9.1",
"cordova-clipboard": "1.3.0",
"cordova-plugin-camera": "5.0.2",
"cordova-plugin-firebase": "2.0.5",
"d3": "7.0.0",
"deep-equal": "1.1.1",
"env-cmd": "10.1.0",
"eslint": "7.26.0",
"fake-indexeddb": "3.1.7",
"firebase": "9.5.0",
"formik": "2.2.9",
"generate-avatar": "1.4.10",
"hash-wasm": "4.9.0",
"history": "4.0.0",
"husky": "6.0.0",
"ionicons": "5.4.0",
"lint-staged": "11.0.0",
"prettier": "2.3.0",
"promise-all-settled-polyfill": "0.1.2",
"query-string": "7.0.1",
"react": "17.0.2",
"react-dom": "17.0.2",
"react-error-boundary": "3.1.3",
"react-geocode": "0.2.3",
"react-image-lightbox": "5.1.4",
"react-redux": "7.2.4",
"react-router": "5.2.0",
"react-router-dom": "5.2.0",
"react-scripts": "4.0.3",
"redux": "4.1.0",
"styled-components": "5.3.0",
"typescript": "4.4.4",
"uuid": "8.3.2",
"web-vitals": "0.2.4",
"workbox-background-sync": "5.1.4",
"workbox-broadcast-update": "5.1.4",
"workbox-cacheable-response": "5.1.4",
"workbox-core": "5.1.4",
"workbox-expiration": "5.1.4",
"workbox-google-analytics": "5.1.4",
"workbox-navigation-preload": "5.1.4",
"workbox-precaching": "5.1.4",
"workbox-range-requests": "5.1.4",
"workbox-routing": "5.1.4",
"workbox-strategies": "5.1.4",
"workbox-streams": "5.1.4"
```

### PR DESCRIPTION
**Pull request by liamdebeasi**
## Pull request checklist

Please check if your PR fulfills the following requirements:
- [ ] Tests for the changes have been added (for bug fixes / features)
- [ ] Docs have been reviewed and added / updated if needed (for bug fixes / features)
- [x] Build (`npm run build`) was run locally and any changes were pushed
- [x] Lint (`npm run lint`) has passed locally and any fixes were made for failures

## Pull request type

 

 

Please check the type of change your PR introduces:
- [x] Bugfix
- [ ] Feature
- [ ] Code style update (formatting, renaming)
- [ ] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] Documentation content changes
- [ ] Other (please describe): 

## What is the current behavior?

Issue Number: resolves #24480

The asset path for Ionicons is not set when using custom elements build. As a result, we cannot lazy load icons internally. We should be using the icon SVG data anyways as it results in additional maintenance overhead (I.e. https://github.com/ionic-team/ionic-framework/blob/main/packages/vue/src/index.ts#L154-L169);

## What is the new behavior?

- All ion-icons used internally in refresher-content are now imported rather than referenced by name.
- Note: "lines" is a spinner not an ion-icon, and the spinner SVG data is defined inside of Ionic (not ionicons) so we can leave that as is: https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/spinner/spinner-configs.ts#L83

## Does this introduce a breaking change?

- [ ] Yes
- [x] No

## Other information

Dev build: 6.0.1-dev.1641502305.cfaa4b3

### CODE REVIEW

File: core/src/components/refresher-content/refresher-content.tsx [hunk 1]
```diff
@@ -51,7 +52,7 @@ export class RefresherContent implements ComponentInterface {
   componentWillLoad() {
     if (this.pullingIcon === undefined) {
       const mode = getIonMode(this);
-      const overflowRefresher = (this.el.style as any).webkitOverflowScrolling !== undefined ? 'lines' : 'arrow-down';
+      const overflowRefresher = (this.el.style as any).webkitOverflowScrolling !== undefined ? 'lines' : arrowDown;
```
Comment (sean-perkins) [inline core/src/components/refresher-content/refresher-content.tsx:55]: Wouldn't `lines` need to be swapped out with the SVG content as well?

Comment (liamdebeasi) [reply core/src/components/refresher-content/refresher-content.tsx:55]: "lines" is a spinner not an ion-icon, and the spinner SVG data is defined inside of Ionic (not ionicons): https://github.com/ionic-team/ionic-framework/blob/main/core/src/components/spinner/spinner-configs.ts#L83

**Review by sean-perkins — APPROVED** [review]

**Review by amandaesmith3 — APPROVED** [review]

**Comment by mamitry** [comment]
In which release did this fix get into?

**Comment by amandaesmith3** [comment]
This was in 6.0.2. You can view release notes with a list of merged PRs here: https://ionicframework.com/docs/reference/release-notes

**Comment by mamitry** [comment]
Thank You

**Pull request merged**